MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -1655,9 +1655,43 @@ SyncBitmapAndSetAccessAttributesInMemoryMap (
     // If attributes are nonzero, set the corresponding bit in the bitmap
     if (MemoryMapEntry->Attribute != 0) {
       SET_BITMAP_INDEX (Bitmap, Index);
+    }
+
+    Index++;
+    MemoryMapEntry = NEXT_MEMORY_DESCRIPTOR (MemoryMapEntry, *DescriptorSize);
+  }
+}
 
-      // Skip code regions
-    } else if (!IsCodeType (MemoryMapEntry->Type)) {
+/**
+  Set access attributes in the memory map based on the memory protection policy and
+  mark visited regions in the bitmap.
+
+  @param[in]        MemoryMapSize       A pointer to the size, in bytes, of the
+                                        MemoryMap buffer.
+  @param[in, out]   MemoryMap           A pointer to the current memory map.
+  @param[in]        DescriptorSize      Size, in bytes, of an individual EFI_MEMORY_DESCRIPTOR.
+  @param[in]        Bitmap              Pointer to the beginning of the bitmap to be updated
+**/
+STATIC
+VOID
+SetAccessAttributesInMemoryMap (
+  IN CONST UINTN             *MemoryMapSize,
+  IN  EFI_MEMORY_DESCRIPTOR  *MemoryMap,

```

Comment:
nit: Should this be `IN OUT`? Or comment changed to `IN` only?

Reply:
Updated to IN OUT :)